src/main/java/org/intellij/sdk/codesync/files/AlertsFile.java

```diff
@@ -0,0 +1,159 @@
+    public Map<String, Object> getYMLAsHashMap() {
+        return contentsMap;
+    }
+
+    public AlertsFile (String filePath) throws FileNotFoundException, InvalidYmlFileError {
+        File alertsFile = new File(filePath);
+
+        if (!alertsFile.isFile()) {
+            throw new FileNotFoundException(String.format("Alerts file \"%s\" does not exist.", filePath));
+        }
+        this.alertsFile = alertsFile;
+        this.contentsMap = this.readYml();
+    }
+
+    private static AlertsFile getInstance() {
+        try {
+            return new AlertsFile(ALERTS_FILE_PATH);
+        } catch (InvalidYmlFileError | FileNotFoundException error) {
+            CodeSyncLogger.error(
+                String.format(
+                    "Error while instantiating AlertsFile instance. Error: %s",
+                    error.getMessage()
+                )
+            );
+        }
+        return null;
+    }
+
+    public static void updateUserActivity(Instant activityInstant) {
+        AlertsFile alertsFile = getInstance();
+        if (alertsFile == null) {
+            return;
+        }
+        alertsFile.contentsMap.put(
+            "user_activity", CommonUtils.formatDate(Date.from(activityInstant), DATE_TIME_FORMAT_WITHOUT_TIMEZONE)
+        );
+
+        try {
+            alertsFile.writeYml();
+        } catch (FileNotFoundException | FileLockedError error) {
+            CodeSyncLogger.error(String.format("Error while writing to alerts.yml file. Error: %s", error.getMessage()));
+        } catch (InvalidYmlFileError error) {
+            CodeSyncLogger.error(
+                String.format(
+                    "Error while writing to alerts.yml file, removing file contents to avoid further errors. Error: %s",
+                    error.getMessage()
+                )
+            );
+            alertsFile.removeFileContents();
+        }
+    }
+
+    public static void updateUpgradePlanActivity(Instant activityInstant) {
+        AlertsFile alertsFile = getInstance();
+        if (alertsFile == null) {
+            return;
+        }
+        alertsFile.contentsMap.put(
+            "upgrade_plan", CommonUtils.formatDate(Date.from(activityInstant), DATE_TIME_FORMAT_WITHOUT_TIMEZONE)
+        );
+
+        try {
+            alertsFile.writeYml();
+        } catch (FileNotFoundException | FileLockedError error) {
+            CodeSyncLogger.error(String.format("Error while writing to alerts.yml file. Error: %s", error.getMessage()));
+        } catch (InvalidYmlFileError error) {
+            CodeSyncLogger.error(
+                String.format(
+                    "Error while writing to alerts.yml file, removing file contents to avoid further errors. Error: %s",
+                    error.getMessage()
+                )
+            );
+            alertsFile.removeFileContents();
+        }
+    }
+    public static void updateTeamActivity(String userEmail, Instant checkedAt, Instant checkedFor, Instant shownAt) {
```

Comment:
Since current API only reflects team-activity, I think for now only `updateTeamActivity` is required, we can keep other functions though and can use them later as use-cases come